contracts/messaging/dfk/HeroCoreUpgradeable.sol

```diff
@@ -0,0 +1,149 @@
+        Hero memory _hero = Hero({
+            id: nextHeroId,
+            info: HeroInfo({
+                statGenes: _statGenes,
+                visualGenes: _visualGenes,
+                rarity: _rarity,
+                shiny: _shiny,
+                generation: _crystal.generation,
+                firstName: _crystal.firstName,
+                lastName: _crystal.lastName,
+                shinyStyle: _crystal.shinyStyle,
+                class: statScience.getGene(_statGenes, 44), // class is position 44
+                subClass: statScience.getGene(_statGenes, 40) // subclass is position 40
+            }),
+            state: HeroState({
+                level: 1,
+                xp: 0,
+                currentQuest: address(0),
+                staminaFullAt: 0,
+                hpFullAt: 0,
+                mpFullAt: 0,
+                sp: 0,
+                status: HeroStatus.OK
+            }),
+            summoningInfo: SummoningInfo({
+                summonedTime: block.timestamp,
+                nextSummonTime: block.timestamp,
+                summonerId: _crystal.summonerId,
+                assistantId: _crystal.assistantId,
+                summons: 0,
+                maxSummons: _crystal.maxSummons
+            }),
+            stats: statScience.generateStats(_statGenes, _crystal, _rarity, _crystalId),
+            primaryStatGrowth: statScience.generateStatGrowth(_statGenes, _crystal, _rarity, true),
+            secondaryStatGrowth: statScience.generateStatGrowth(_statGenes, _crystal, _rarity, false),
+            professions: HeroProfessions({mining: 0, gardening: 0, foraging: 0, fishing: 0})
+        });
+        heroes[nextHeroId] = _hero;
+        nextHeroId++;
+        // emit the summon event
+        emit HeroSummoned(
+            _crystal.owner,
+            _hero.id,
+            uint256(_hero.summoningInfo.summonerId),
+            uint256(_hero.summoningInfo.assistantId),
+            _hero.info.statGenes,
+            _hero.info.visualGenes
+        );
+        // Send the newly created hero to the owner.
+        _mint(_crystal.owner, _hero.id);
+        return _hero.id;
+    }
+    /// @dev Saves a hero object to storage.
+    function updateHero(Hero memory _hero) external onlyRole(HERO_MODERATOR_ROLE) whenNotPaused {
+        // Save the hero.
+        heroes[_hero.id] = _hero;
+        emit HeroUpdated(ownerOf(_hero.id), _hero.id, _hero);
+    }
+    function bridgeMint(uint256 _id, address _to) external onlyRole(BRIDGE_ROLE) {
+        _mint(_to, _id);
+    }
+    // /**
+    //  * @dev See {IERC165-supportsInterface}.
+    //  */
+    // /// TODO find out if this is right, Im not sure
+    function supportsInterface(bytes4 interfaceId) public view override(ERC721EnumerableUpgradeable, AccessControlUpgradeable) returns (bool) {
+        // return interfaceId == type(IHeroTypes).interfaceId || super.supportsInterface(interfaceId);
```

Comment:
```suggestion
        return interfaceId == type(IHeroTypes).interfaceId || super.supportsInterface(interfaceId);
```
This should be uncommented, otherwise there's no sense in overriding the function.